element_calcium_imaging/imaging_report.py

```diff
@@ -0,0 +1,418 @@
+import cv2
+import numpy as np
+import datajoint as dj
+from functools import partial
+from plotly.io import from_json
+import plotly.graph_objects as go
+from ipywidgets import widgets as wg
+from . import imaging, scan
+
+schema = imaging.schema
+
+
+def single_to_3channel_image(image, low_q, high_q):
+    low_p, high_p = np.percentile(image, [low_q, high_q])
+    image = np.clip(image, low_p, high_p)
+
+    image -= image.min()
+    image *= 255 / image.max()
+    image = np.uint8(image)
+    image = np.dstack([image] * 3)
+    return image
+
+
+def paint_rois(image, mask_xpix, mask_ypix):
+    # Generate random hsv colors
+    SATURATION = 40
+    VALUE = 255
+    hues = np.random.sample(size=len(mask_xpix)) * 10000
+    hsv_colors = np.stack(
+        [hues, np.full(len(mask_xpix), SATURATION), np.full(len(mask_xpix), VALUE)]
+    ).T
+
+    # Assign colors to each region
+    masks = cv2.cvtColor(image, cv2.COLOR_RGB2HSV)
+    for xpix, ypix, hsv_color in zip(mask_xpix, mask_ypix, hsv_colors):
+        masks[ypix, xpix] = hsv_color
+
+    masks = np.uint8(cv2.cvtColor(masks.astype(np.float32), cv2.COLOR_HSV2RGB))
+    return masks
+
+
+def alpha_combine_2images(image, masks):
+    alpha = 0.5
+    return np.uint8(image * alpha + masks * (1 - alpha))
+
+
+def make_maskid_image(img, roi_ids, mask_xpix, mask_ypix):
+    shape = img.shape[:2]
+    maskid_image = np.full(shape, -1)
```

Comment:
Do you want to use `nan` instead of -1

Reply:
The values of the maskid_image are used in the plotly's cursor tooltip (A box shows the values attached to each pixel. Here we show the mask ids). I need to check if plotly supports nan. If does, I can switch to `nan`.

Reply:
Looks like storing np.nan with integers is problematic. It's only supported in lataest versions of the numpy. We would probably have version conflicts (e.g. suite2p or caiman has a specific numpy version requirement).